### PR TITLE
feat(validation): list invalid fields for the action-button widget

### PR DESCRIPTION
**Pull request by JohnL17**
Add the possibility to show invalid fields for the action-button widget.

Depends on https://github.com/projectcaluma/caluma/pull/2050

### CODE REVIEW

File: packages/-ember-caluma/app/templates/demo/form-rendering.hbs [hunk 1]
```diff
@@ -1,6 +1,9 @@
 {{#if @model}}
   <CfContent
     @documentId={{@model}}
-    @context={{hash actionButtonOnSuccess=this.actionButtonOnSuccess}}
+    @context={{hash
+      actionButtonOnSuccess=this.actionButtonOnSuccess
+      showValidation=false
```
Comment (czosel) [inline packages/-ember-caluma/app/templates/demo/form-rendering.hbs:6]: `false` is the default, you can remove this line.

File: packages/form/translations/de.yaml [hunk 2]
```diff
@@ -51,3 +51,6 @@ caluma:
       uploadFailed: "Beim Hochladen ist ein Fehler aufgetreten."
       format: "{errorMsg}"
       table: "Mindestens eine Zeile der Tabelle wurde nicht korrekt ausgefüllt"
+      error: "Folgende Fragen sind noch nicht korrekt ausgefüllt:"
+      success: "Alle Fragen wurden korrekt ausgefüllt!"
```
Comment (czosel) [inline packages/form/translations/de.yaml:55]: you can drop this as well, since it's not used

File: packages/form/translations/en.yaml [hunk 3]
```diff
@@ -51,3 +51,6 @@ caluma:
       uploadFailed: "An error occured during upload."
       format: "{errorMsg}"
       table: "At least one row of the table was not filled in correctly"
+      error: "The following questions have not yet been filled in correctly:"
+      success: "All questions have been filled in correctly!"
```
Comment (czosel) [inline packages/form/translations/en.yaml:55]: same here

File: packages/form/translations/de.yaml [hunk 4]
```diff
@@ -51,3 +51,6 @@ caluma:
...
+      loading: "Ihr Formular wird geprüft. Bitte haben Sie einen Moment Geduld."
```
Comment (czosel) [inline packages/form/translations/de.yaml:56]: you can drop `success` and `loading`, since they're not used

**Comment by czosel** [comment]
@anehx I wasn't able to find any documentation for `<CfContent>` - did we never have any?

File: packages/form/translations/de.yaml (hunk 4 above)
Comment (JohnL17) [reply packages/form/translations/de.yaml]: Done.

File: packages/-ember-caluma/app/templates/demo/form-rendering.hbs (hunk 1 above)
Comment (JohnL17) [reply packages/-ember-caluma/app/templates/demo/form-rendering.hbs]: Removed it.

File: packages/form/addon/components/cf-field/input/action-button.hbs [hunk 5]
```diff
@@ -8,6 +8,24 @@
     @validateOnEnter={{this.validateOnEnter}}
     as |isValid validate|
   >
+    {{#if (and this.invalidFields.length @context.showValidation)}}
```
Comment (anehx) [inline packages/form/addon/components/cf-field/input/action-button.hbs:11]: As discussed with @czosel: `@context.showValidation` will be replaced by a proper backend configuration property similar to `validateOnEnter`

File: packages/form/addon/components/cf-field/input/action-button.js [hunk 6]
```diff
@@ -12,8 +12,9 @@ if (macroCondition(dependencySatisfies("@projectcaluma/ember-workflow", ""))) {
       super(...args);
 
       assert(
-        "The document must have a `workItemUuid` for `<CfField::Input::ActionButton />` to work.",
-        this.args.field.document.workItemUuid,
+        "The document or context must have a `workItem` for `<CfField::Input::ActionButton />` to work.",
+        this.args.field.document.workItemUuid ||
+          this.args.context?.actionButtonWorkItemId,
```
Comment (anehx) [inline packages/form/addon/components/cf-field/input/action-button.js:17]: Assertion message must be more explicit to be helpful

File: packages/form/translations/fr.yaml [hunk 7]
```diff
@@ -51,3 +51,4 @@ caluma:
       uploadFailed: "Une erreur s'est produite pendant le téléchargement."
       format: "{errorMsg}"
       table: "Au moins une ligne du tableau n'a pas été remplie correctement"
+      error: "Les questions suivantes ne sont pas encore correctement remplies :"
```
Comment (anehx) [inline packages/form/translations/fr.yaml:54]:
```suggestion
      error: "Les questions suivantes n'ont pas encore été correctement remplies :"
```

**Review by anehx — CHANGES_REQUESTED** [review]

File: packages/form/addon/components/cf-field/input/action-button.hbs (hunk 5 above)
Comment (czosel) [reply packages/form/addon/components/cf-field/input/action-button.hbs]: done :+1:

File: packages/form/translations/fr.yaml (hunk 7 above)
Comment (czosel) [reply packages/form/translations/fr.yaml]: fixed

File: packages/form/addon/components/cf-field/input/action-button.js (hunk 6 above)
Comment (czosel) [reply packages/form/addon/components/cf-field/input/action-button.js:17]: I tried to improve it a bit.

**Review by anehx — CHANGES_REQUESTED** [review]
Only nitpicking, otherwise LGTM :100: 

Also, much :heart: for writing a test!!

File: packages/form/addon/components/cf-field/input/action-button.js [hunk 8]
```diff
@@ -39,6 +40,16 @@ if (macroCondition(dependencySatisfies("@projectcaluma/ember-workflow", ""))) {
       );
     }
 
+    get showValidation() {
+      return this.args.field.question.raw.showValidation;
+    }
```
[the remark this answers is not in the document]
Comment (czosel) [reply packages/form/addon/components/cf-field/input/action-button.js]: fair, fixed!

**Review by anehx — APPROVED** [review]

**Pull request merged**